precompiles/staking/tx.go

```diff
@@ -81,14 +84,65 @@ func (p Precompile) CreateValidator(
 		return nil, err
 	}
 
-	// Emit the event for the delegate transaction
+	// Emit the event for the create validator transaction
 	if err = p.EmitCreateValidatorEvent(ctx, stateDB, msg, delegatorHexAddr); err != nil {
 		return nil, err
 	}
 
 	return method.Outputs.Pack(true)
 }
 
+// EditValidator performs edit validator.
+func (p Precompile) EditValidator(
+	ctx sdk.Context,
+	origin common.Address,
+	_ *vm.Contract,
+	stateDB vm.StateDB,
+	method *abi.Method,
+	args []interface{},
+) ([]byte, error) {
+	msg, validatorHexAddr, err := NewMsgEditValidator(args)
+	if err != nil {
+		return nil, err
+	}
+
+	commissionRate := strconv.Itoa(DoNotModifyCommissionRate)
+	if msg.CommissionRate != nil {
+		commissionRate = msg.CommissionRate.String()
+	}
+
+	minSelfDelegation := strconv.Itoa(DoNotModifyMinSelfDelegation)
+	if msg.CommissionRate != nil {
+		commissionRate = msg.MinSelfDelegation.String()
```

Comment:
```suggestion
		minSelfDelegation = msg.MinSelfDelegation.String()
```

Reply:
I copied the above code and forgot to change it, hahaha.